### ISSUE TITLE
"gomplate -in flibbit" should produce an error

### ISSUE THREAD

**Issue by docwhat** [hairyhenderson/gomplate#192]
I would expect that non-flag arguments would raise an error in `gomplate`.

Expected:

``` sh
$ gomplate -in flibbit
I don't know what to do with 'flibbit'
# non-zero exit code
$
```

Actual:

``` sh
$ gomplate -in flibbit
n<no newline> 
# zero exit code
$
```

I got bit by this because I typed `-in` instead of `--in` and spent a while trying to figure out what I did wrong.

FYI: `-in` can be rewritten as `--in 'n'`

**Comment by hairyhenderson** [comment]
Thanks for reporting this, @docwhat! I agree - that's a bit odd. This _used_ to be the case, but I think additional args are silently ignored ever since I moved to using `cobra` for parsing args.

**Comment by docwhat** [comment]
When I was using kingpin I had a similar problem; I was not checking all the error conditions.

**Comment by docwhat** [comment]
Reading [the docs on arguments](https://github.com/spf13/cobra#positional-and-custom-arguments), I _think_ what you want is a line in the [`RootCmd` section](https://github.com/hairyhenderson/gomplate/blob/b15337e66361b6839d50684ea78fad8df7c0bc17/main.go#L60) saying something like:

``` go
...
Args: NoArgs,
...
```

But I'm not sure, since I'm just speed reading the docs and change history of cobra.

**Comment by docwhat** [comment]
@hairyhenderson 

P.S. For macOS users Homebrew/homebrew-core#15811

**Comment by hairyhenderson** [comment]
thanks @docwhat - I already have a tap at https://github.com/hairyhenderson/homebrew-tap, but being in core would be cool!

**Comment by drmdrew** [comment]
@hairyhenderson this looks to have a quick fix so I submitted PR #208 that registers cobra.NoArgs for the rootCmd as mentioned by @docwhat earlier https://github.com/hairyhenderson/gomplate/issues/192#issuecomment-320481858
